### PR TITLE
Chapter18

### PR DESCRIPTION
**Pull request by nazihaltajar**

### CODE REVIEW

File: Chapter9/Chapter9/Controller/NewRestaurantController.swift [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            navigationController?.navigationBar.largeTitleTextAttributes = [
+                NSAttributedString.Key.foregroundColor: UIColor.customColor,
+                NSAttributedString.Key.font: customFont]
+        }
+    }
+}
+
+extension NewRestaurantController: UITextFieldDelegate {
+
+func textFieldShouldReturn(_ textField: UITextField) -> Bool {
```
Comment (cristibaluta) [inline Chapter9/Chapter9/Controller/NewRestaurantController.swift]: Wrong indentation

File: Chapter9/Chapter9/Controller/NewRestaurantController.swift [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            }
+            present(photoSourceRequestController, animated: true, completion: nil)
+        }
+    }
+}
+
+extension NewRestaurantController: UIImagePickerControllerDelegate {
+
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey: Any]) {
+     if let selectedImage = info[ UIImagePickerController.InfoKey.originalImage ] as? UIImage {
```
Comment (cristibaluta) [inline Chapter9/Chapter9/Controller/NewRestaurantController.swift]: Wrong indentation

File: Chapter9/Chapter9/Controller/NewRestaurantController.swift [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+//  NewRestaurantController.swift
+//  Chapter9
+//
+//  Created by Nazih Al Tajar on 08/04/2019.
+//  Copyright © 2019 sss. All rights reserved.
+//
+
+import UIKit
+
+class NewRestaurantController: UITableViewController, UINavigationControllerDelegate {
```
Comment (adinas23) [inline Chapter9/Chapter9/Controller/NewRestaurantController.swift]: ```NewRestaurantViewController```

File: Chapter9/Chapter9/Controller/NewRestaurantController.swift [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    @IBOutlet weak var photoImageView: UIImageView!
+    @IBOutlet weak var nameTextField: RoundedTextFiled! {
+        didSet {
+            nameTextField.tag  = 1
```
Comment (adinas23) [inline Chapter9/Chapter9/Controller/NewRestaurantController.swift]: You can set it from storyboard

File: Chapter9/Chapter9/Controller/NewRestaurantController.swift [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+//  Chapter9
+//
+//  Created by Nazih Al Tajar on 08/04/2019.
+//  Copyright © 2019 sss. All rights reserved.
+//
+
+import UIKit
+
+class NewRestaurantController: UITableViewController, UINavigationControllerDelegate {
+    @IBOutlet weak var photoImageView: UIImageView!
```
Comment (adinas23) [inline Chapter9/Chapter9/Controller/NewRestaurantController.swift]: weak private

File: Chapter9/Chapter9/Controller/NewRestaurantController.swift [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    @IBOutlet weak var nameTextField: RoundedTextFiled! {
```
Comment (adinas23) [inline Chapter9/Chapter9/Controller/NewRestaurantController.swift]: weak private

File: Chapter9/Chapter9/Controller/NewRestaurantController.swift [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+        didSet {
+            nameTextField.tag  = 1
+            nameTextField.delegate = self
+        }
+    }
+    @IBOutlet weak var typeTextField: RoundedTextFiled! {
```
Comment (adinas23) [inline Chapter9/Chapter9/Controller/NewRestaurantController.swift]: private

File: Chapter9/Chapter9/Controller/NewRestaurantController.swift [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+        if let customFont = UIFont(name: "Rubik-Medium", size: 35.0) {
+            navigationController?.navigationBar.largeTitleTextAttributes = [
+                NSAttributedString.Key.foregroundColor: UIColor.customColor,
+                NSAttributedString.Key.font: customFont]
+        }
+    }
+}
+
+extension NewRestaurantController: UITextFieldDelegate {
+
```
Comment (adinas23) [inline Chapter9/Chapter9/Controller/NewRestaurantController.swift]: Remove extra line

File: Chapter9/Chapter9/Controller/NewRestaurantController.swift [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+    if let nextTextFiled = view.viewWithTag(textField.tag + 1) {
+        textField.resignFirstResponder()
+        nextTextFiled.becomeFirstResponder()
+    }
+    return true
+    }
+}
+
+extension NewRestaurantController {
+
```
Comment (adinas23) [inline Chapter9/Chapter9/Controller/NewRestaurantController.swift]: Remove extra line

File: Chapter9/Chapter9/Controller/NewRestaurantController.swift [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+                }
+            })
+            photoSourceRequestController.addAction(cameraAction)
+            photoSourceRequestController.addAction(photoLibraryAction)
+
+            if let popoverController = photoSourceRequestController.popoverPresentationController {
+                if let cell = tableView.cellForRow(at: indexPath) {
+                    popoverController.sourceView = cell
+                    popoverController.sourceRect = cell.bounds
+                    popoverController.permittedArrowDirections = .any
```
Comment (adinas23) [inline Chapter9/Chapter9/Controller/NewRestaurantController.swift]: You can remove this line. Not needed anymore

File: Chapter9/Chapter9/Controller/NewRestaurantController.swift [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+                }
+            }
+            present(photoSourceRequestController, animated: true, completion: nil)
+        }
+    }
+}
+
+extension NewRestaurantController: UIImagePickerControllerDelegate {
+
```
Comment (adinas23) [inline Chapter9/Chapter9/Controller/NewRestaurantController.swift]: Remove extra line

File: Chapter9/Chapter9/Controller/NewRestaurantController.swift [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey: Any]) {
+     if let selectedImage = info[ UIImagePickerController.InfoKey.originalImage ] as? UIImage {
+            photoImageView.image = selectedImage
+            photoImageView.contentMode = .scaleToFill
+            photoImageView.clipsToBounds = true
+        }
+        let leadingContraint = NSLayoutConstraint(item: photoImageView ?? "", attribute: .leading, relatedBy: .equal,
```
Comment (adinas23) [inline Chapter9/Chapter9/Controller/NewRestaurantController.swift]: ```leadingConstraint```. You missed a "s"

File: Chapter9/Chapter9/Controller/RestaurantTableViewController.swift [hunk 13]
```diff
@@ -19,6 +19,9 @@ protocol CustomCell: class {
 
 class RestaurantTableViewController: UIViewController {
     @IBOutlet private weak var tableView: UITableView!
+    @IBAction func unwindToHome(segue: UIStoryboardSegue) {
```
Comment (adinas23) [inline Chapter9/Chapter9/Controller/RestaurantTableViewController.swift]: Make the method private

File: Chapter9/Chapter9/View/RoundedTextFiled.swift [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+//  Chapter9
+//
+//  Created by Nazih Al Tajar on 08/04/2019.
+//  Copyright © 2019 sss. All rights reserved.
+//
+
+import UIKit
+
+class RoundedTextFiled: UITextField {
+
```
Comment (adinas23) [inline Chapter9/Chapter9/View/RoundedTextFiled.swift]: Remove extra line

File: Chapter9/Chapter9/View/RoundedTextFiled.swift [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+    let padding = UIEdgeInsets(top: 0, left: 10, bottom: 0, right: 10)
+
+    override func textRect(forBounds bounds: CGRect) -> CGRect {
+        return bounds.inset(by: padding)
+    }
+    override func placeholderRect(forBounds bounds: CGRect) -> CGRect {
```
Comment (adinas23) [inline Chapter9/Chapter9/View/RoundedTextFiled.swift:18]: Put an extra line

**Pull request closed without merging**